go-controller/pkg/util/util.go

```diff
@@ -323,3 +324,46 @@ func UpdateNodeSwitchExcludeIPs(nbClient libovsdbclient.Client, nodeName string,
 
 	return nil
 }
+
+// NetNameInfo is structure which holds network name information
+type NetNameInfo struct {
+	// netconf's name, default for default network
+	NetName string
+	// Prefix of OVN logical entities for this network
+	Prefix      string
+	IsSecondary bool
+}
+
+// NetAttachDefInfo is structure which holds specific per-network information
+type NetAttachDefInfo struct {
+	NetNameInfo
+	NetCidr string
+	MTU     int
+}
+
+// Note that for port_group and address_set, it does not allow the '-' character
+// Also replace "/" in nadName with "."
+func GetNetworkPrefix(netName string, isDefault bool) string {
+	if isDefault {
+		return ""
+	}
+	name := strings.ReplaceAll(netName, "-", ".")
+	name = strings.ReplaceAll(name, "/", ".")
+	return name + "_"
+}
+
+func NewNetAttachDefInfo(netconf *cnitypes.NetConf) (*NetAttachDefInfo, error) {
```

Comment:
there could be error case when we introduce non-default network, as we'd need to validate non-default network netconf.

Reply:
OK I see. I would expect the function's signature to change when said changes are introduced, since **right now** there's absolutely no reason for this function to return an error - the code it has encapsulated just can't fail.

Reply:
but this MR is preparing for the later introduction of secondary network support, the whole secondary network controller is not used right now.

Reply:
Yes, but it makes life harder for the reviewer: there's no way for me to know what you'll do in X PRs down the line; when I see something that doesn't make sense / I don't understand, I need to raise it. 

The answer "it'll make sense when the rest of the code is added" is, not the best IMO: the code should make sense in a commit by commit basis. In this PR, returning the error can be avoided: there's no way for that code to produce an error.

... when the code base changes and you introduce the code that can throw an error, then would be the right time to update the funcs signature.

Anyway, I won't choose this hill to die on. You can leave it as is if you prefer, now that I understand why it is as it is.